File: src/Cryptofolio.App/ClientApp/src/app/api-options.ts
import { Injectable } from "@angular/core";

@Injectable()
export class ApiOptions {
  private _url: string;
  public get url() {
    return this._url;
  }

  private _assetsEndpoint: string;
  public get assetsEndpoint() {
    return this._assetsEndpoint;
  }

  private _currenciesEndpoint: string;
  public get currenciesEndpoint() {
    return this._currenciesEndpoint;
  }

  private _exchangesEndpoint: string;
  public get exchangesEndpoint() {
    return this._exchangesEndpoint;
  }

  private _transactionsEndpoint: string;
  public get transactionsEndpoint() {
    return this._transactionsEndpoint;
  }
  
  private _walletsEndpoint: string;
  public get walletsEndpoint() {
    return this._walletsEndpoint;
  }

  constructor() {
    const apiOptions = (window as any)['__me']['api'];
    this._url = apiOptions.url;
    this._assetsEndpoint = apiOptions.assetsEndpoint;
    this._currenciesEndpoint = apiOptions.currenciesEndpoint;
    this._exchangesEndpoint = apiOptions.exchangesEndpoint;
    this._transactionsEndpoint = apiOptions.transactionsEndpoint;
    this._walletsEndpoint = apiOptions.walletsEndpoint;
  }
}

File: src/Cryptofolio.App/ApiOptions.cs
namespace Cryptofolio.App
{
    /// <summary>
    /// Configuration options for the API endpoints.
    /// </summary>
    public class ApiOptions
    {
        /// <summary>
        /// The API url.
        /// </summary>
        public string Url { get; set; }

        /// <summary>
        /// The assets endpoint.
        /// </summary>
        public string AssetsEndpoint => $"{Url}/assets";

        /// <summary>
        /// The currencies endpoint.
        /// </summary>
        public string CurrenciesEndpoint => $"{Url}/currencies";

        /// <summary>
        /// The exchanges endpoint.
        /// </summary>
        public string ExchangesEndpoint => $"{Url}/exchanges";

        /// <summary>
        /// The transactions endpoint.
        /// </summary>
        public string TransactionsEndpoint => $"{Url}/transactions";

        /// <summary>
        /// The wallets endpoint.
        /// </summary>
        public string WalletsEndpoint => $"{Url}/wallets";

    }
}
